r/vignettes/developers/bindings.Rmd

```diff
@@ -0,0 +1,238 @@
+---
+title: "Writing Bindings"
+---
+
+```{r, include=FALSE}
+library(arrow, warn.conflicts = FALSE)
+library(dplyr, warn.conflicts = FALSE)
+```
+
+
+When writing bindings between C++ compute functions and R functions, the aim is 
+to expose the C++ functionality via existing R functions.  The syntax and 
+functionality should (usually) exactly match that of the existing R functions 
+(though with some exceptions) so that users are able to use existing tidyverse 
+or base R syntax, or call existing S3 methods on objects, whilst taking 
+advantage of the speed and functionality of the underlying arrow package.
```

Comment:
```suggestion
When writing bindings between C++ compute functions and R functions, the aim is 
to expose the C++ functionality via existing R functions. The syntax and 
functionality should match that of the existing R functions 
(though with some exceptions) so that users are able to use existing tidyverse 
or base R syntax, or call existing S3 methods on objects, whilst taking 
advantage of the speed and functionality of the underlying arrow package.
```
This intro is concise, I like it!
I would suggest to remove "(usually) exactly" as it is slightly confusing (should or shouldn't then) and for me "match" together with the second part "(though with some exceptions)" makes it clear.

Reply:
Excellent point!